packages/web-frontend/src/containers/EnlargedDialog/EnlargedDialog.js

```diff
@@ -88,6 +80,66 @@ const hasDrillDownChanged = (options, cachedOptions) =>
   options.parameterLink !== cachedOptions.parameterLink ||
   options.parameterValue !== cachedOptions.parameterValue;
 
+const useDrillDownState = contentByLevel => {
+  const [drillDownState, setDrillDownState] = useState({
+    drillDownLevel: 0,
+    parameterLinks: {},
+    parameterValues: {},
+  });
+
+  // Regardless of the drillDown level, we pass the base view content through
+  const baseViewContent = contentByLevel?.[0]?.viewContent;
+  const drillDownContent = contentByLevel?.[drillDownState.drillDownLevel]?.viewContent;
+  const baseViewConfig = contentByLevel?.[0]?.viewConfig;
+  const drillDownConfig = contentByLevel?.[drillDownState.drillDownLevel]?.viewConfig;
+
+  const viewContent =
+    baseViewContent === undefined ? null : { ...baseViewConfig, ...baseViewContent };
+  const newDrillDownContent =
+    drillDownContent === undefined ? null : { ...drillDownConfig, ...drillDownContent };
+
+  return {
+    drillDownConfig,
+    drillDownState,
+    viewContent,
+    newDrillDownContent,
```

Comment:
Seems like the `new` part is just for convenience in this function because we've already used the name we want, but the hook should probably just expose `drillDownContent` right?
```suggestion
    drillDownContent: newDrillDownContent,
```